### PR TITLE
C#: Improve log messages in buildless mode + some cleanup/refactoring

### PR DESCRIPTION
**Pull request by tamasvajk**
This pull request mainly focuses on improving the log messages written during buildless extraction. Additionally code is refactored to remove the `ProgressMonitor` class and instead use the logger directly. Furthermore, the `dotnet restore` calls are also refactored to reduce the number of (`out`) parameter of the corresponding methods. 

Commit-by-commit review is suggested.

### CODE REVIEW

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs [hunk 1]
```diff
@@ -102,33 +103,36 @@
...
-                AllSourceFiles.Count(),
-                ProjectSourceFiles.Count(),
-                MissingSourceFiles.Count(),
-                ReferenceFiles.Count(),
-                UnresolvedReferences.Count(),
-                conflictedReferences,
-                succeededProjects + failedProjects,
-                failedProjects,
-                DateTime.Now - startTime);
+            const int align = 6;
```
[the remark this answers is not in the document]
Comment (hvitved) [reply csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs:124]: I'll investigate this FP.

[the remark this answers is not in the document]
Comment (michaelnebel) [reply csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs:124]: There is already an issue for this - I will find it.

[the remark this answers is not in the document]
Comment (michaelnebel) [reply csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs:124]: https://github.com/github/codeql-csharp-team/issues/374

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs [hunk 2]
```diff
@@ -595,23 +609,31 @@ private void ResolveConflicts(IEnumerable<string> frameworkPaths)
...
-            foreach (var proj in projectFiles)
+            Parallel.ForEach(solutions, new ParallelOptions { MaxDegreeOfParallelism = options.Threads }, solutionFile =>
             {
-                AnalyseProject(proj);
-            }
+                try
+                {
+                    var sln = new SolutionFile(solutionFile);
+                    progressMonitor.LogInfo($"Analyzing {solutionFile}...");
+                    foreach (var proj in sln.Projects.Select(p => new FileInfo(p)))
```
Comment (michaelnebel) [inline csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs:635]: It looks like the filter `.Where(p => p.Exists)` was removed as a part of the re-factor. Is that intentional?

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs [hunk 3]
```diff
@@ -162,9 +174,9 @@ private void RestoreNugetPackages(List<FileInfo> allNonBinaryFiles, IEnumerable<
                 nugetPackageDllPaths.ExceptWith(excludedPaths);
                 dllPaths.UnionWith(nugetPackageDllPaths);
             }
-            catch (FileNotFoundException)
+            catch (Exception)
```
[the remark this answers is not in the document]
Comment (tamasvajk) [reply csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs:179]: It's only for safety. There's a lot going on in this `try-catch`, for example we're enumerating files to find `packages.config` files, and that logic could potentially throw many exceptions.

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs (hunk 2 above)
Comment (tamasvajk) [reply csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs:635]: `AnalyseProject` checks and checked for the existence of the files. Now if they don't exist, we'll log a message.

**Review by michaelnebel — APPROVED** [review]
This is really excellent!
Very nice improvement of the code quality and the information being logged!

**Pull request merged**